plugins/module_utils/foreman_helper.py

```diff
@@ -697,13 +721,23 @@ def _update_entity(self, resource, desired_entity, current_entity, params, forem
         desired_entity = _flatten_entity(desired_entity, foreman_spec)
         current_entity = _flatten_entity(current_entity, foreman_spec)
         for key, value in desired_entity.items():
+            foreman_type = foreman_spec[key].get('type', 'str')
+            new_value = value
+            old_value = current_entity.get(key)
             # String comparison needs extra care in face of unicode
-            if foreman_spec[key].get('type', 'str') == 'str':
-                if to_native(current_entity.get(key)) != to_native(value):
-                    payload[key] = value
-            else:
-                if current_entity.get(key) != value:
-                    payload[key] = value
+            if foreman_type == 'str':
+                old_value = to_native(current_entity.get(key))
+                new_value = to_native(value)
+            elif foreman_type == 'list':
+                if value and isinstance(value[0], dict):
```

Comment:
Is this something, the `foreman_spec` could tell us?

Reply:
you mean via `elements='dict'`? yeah, maybe, good idea.

Reply:
Well, no, not easily. *tries to wrap my head around that*

Reply:
So the `elements` that Ansible sees are not the same we send to the API. At least not for things like `entity_list` :(